File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.c
int* sortedSquares(int* nums, int numsSize, int* returnSize){
    int i,j,tmp;
    int *k;
    k = (int*)malloc(sizeof(int) * numsSize);
    
    for(i = 0; i < numsSize; ++i)
        k[i] = nums[i]*nums[i];
    
    for(i = 0; i < numsSize-1; ++i)
    {
        j = i+1;
        for(; j < numsSize; ++j)
        {
            if(k[i] > k[j])
            {
                tmp = k[i];
                k[i] = k[j];
                k[j] = tmp;
            }
        }
    }
    
    *returnSize = numsSize;
    return k;
}

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int[] result = new int[nums.length];
        int l = 0;
        int r = nums.length-1;
        int i = nums.length-1;
        
        while(l <= r){
            int n1 = nums[r]*nums[r];
            int n2 = nums[l]*nums[l];
            
            if(n1 > n2){
                result[i] = n1;
                i--;
                r--;
            }
            else{
                result[i] = n2;
                i--;
                l++;
            }
        }
        return result;
    }
}